app/models/form526_submission.rb

```diff
@@ -6,10 +6,82 @@
 
 class Form526Submission < ApplicationRecord
   extend Logging::ThirdPartyTransaction::MethodWrapper
-
+  include AASM
   include SentryLogging
   include Form526ClaimFastTrackingConcern
 
+  # Documentation:
+  # https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/implementation/form_526_state_machine.md
+  aasm do
+    after_all_transitions :log_status_change
+
+    state :unprocessed, initial: true
+    state :delivered_to_primary, :failed_primary_delivery, :rejected_by_primary,
+          :delivered_to_backup, :failed_backup_delivery, :rejected_by_backup,
+          :in_remediation, :finalized_as_successful, :unprocessable
+
+    # - a submission has been delivered to our happy path
+    # - requires polling to finalize
+    event :deliver_to_primary do
+      transitions to: :delivered_to_primary
```

Comment:
Should the state from which an event transitions be declared, or is that not necessary as it's always from unprocessed?

Reply:
typically yes, however (i think this is the doc) this is not a true state machine.  we are allowing any state transition for MVP.  This is intentional to 
1. avoide unintended concequences
2. allow us freedom in remediation, which can be unpredictable.
IF / when this becomes a real state machine we can add that